src/rulesets/oas3/valid-example-ruleset.json

```diff
@@ -0,0 +1,17 @@
+{
```

Comment:
I've never seen a rule specific ruleset before. What is in the intention? Is this for testing?

Reply:
It's to move out this rule from `oas3 rules`. My understanding is that `spectral` can validate jsons that are not `oas2`/`oas3`. We could also move this rule to some other `not oas2/oas3` ruleset, but haven't found one (is there one existing?)

Reply:
functions are generic and can be used anywhere. We have `schemaPath` as a generic function, and this is the basis of the valid example rules.

The rules are different for oas2 and oas3 because the specification structures are different.

So we reuse the schemaPath function, but do not reuse the valid example stuff because we cannot.

Reply:
And that's why I have moved `valid-example` out of `ruleset.json` for oas3

Reply:
Please remove this new file.

Reply:
@philsturgeon, if I remove, where I put `valid-example`? I'm kinda confused. I still have to test `valid-example` rule..

Reply:
@philsturgeon, I can move already existing tests from `src/rulesets/oas3/__tests__/valid-example.ts` to  `src/rulesets/oas2/__tests__/valid-example.ts`, since `valid-example` rule didn't change for `oas2`. So this way `src/rulesets/oas3/__tests__/valid-example.ts` and `src/rulesets/oas3/valid-example-ruleset.json` are removed. Would that be OK?

Reply:
So after seeing the talk about how many different types of param/header/schema example there is, maybe you could use that to create some tests which highlight each type of parameter and property example we looked at, then make sure this change works with each of those? 

The ruleset you added is not the way to go about DRYing things, although I do understand the urge. Mixing up OAS2 and OAS3 is also not the way to go, as the two specs are different. 

Seeing as you have split the `valid-example` rule in two, and the describe tests split in two, I was expecting you to just split the `valid-example` file in two. 

instead of having `describe('valid-schema-example', () => {` inside `src/rulesets/oas3/__tests__/valid-example.ts ` the convention is to match the test file with whatever you are describing, so if you are gonna have two different rules have two different test files.

Is that all starting to make sense?